src/tech-record/TechRecordService.ts

```diff
@@ -0,0 +1,128 @@
+import { Service } from 'typedi';
+import { IWeightDetails } from '../models';
+import { VEHICLE_TYPES } from '../models/Enums';
+import { HTTPError } from '../models/HTTPError';
+import { ISearchResult, TechRecordGet, TechRecordType } from '../models/Types';
+import { TechRecordRepository } from './TechRecordRepository';
+
+@Service()
+export class TechRecordService {
+	constructor(private techRecordRepository: TechRecordRepository) {}
+
+	/**
+	 * Method for getting make and model based on the vehicle from a test-result
+	 * @param testResult - the testResult for which the tech record search is done for
+	 */
+	public getVehicleMakeAndModel = async (testResult: any) => {
+		const searchRes = await this.techRecordRepository.callSearchTechRecords(testResult.systemNumber);
+		const techRecord = await this.processGetCurrentProvisionalRecords(searchRes);
+		// Return bodyMake and bodyModel values for PSVs
+		return techRecord?.techRecord_vehicleType === VEHICLE_TYPES.PSV
+			? {
+					Make: (techRecord as TechRecordType<'psv'>).techRecord_chassisMake,
+					Model: (techRecord as TechRecordType<'psv'>).techRecord_chassisModel,
+				}
+			: {
+					Make: (techRecord as TechRecordType<'hgv' | 'trl'>).techRecord_make,
+					Model: (techRecord as TechRecordType<'hgv' | 'trl'>).techRecord_model,
+				};
+	};
+
+	private processGetCurrentProvisionalRecords = async <T extends TechRecordGet['techRecord_vehicleType']>(
+		searchResult: ISearchResult[]
+	): Promise<TechRecordType<T> | undefined> => {
+		if (searchResult) {
+			const processRecordsRes = this.groupRecordsByStatusCode(searchResult);
+			return processRecordsRes.currentCount !== 0
+				? this.techRecordRepository.callGetTechRecords(
+						processRecordsRes.currentRecords[0].systemNumber,
+						processRecordsRes.currentRecords[0].createdTimestamp
+					)
+				: processRecordsRes.provisionalCount === 1
+					? this.techRecordRepository.callGetTechRecords(
+							processRecordsRes.provisionalRecords[0].systemNumber,
+							processRecordsRes.provisionalRecords[0].createdTimestamp
+						)
+					: this.techRecordRepository.callGetTechRecords(
+							processRecordsRes.provisionalRecords[1].systemNumber,
+							processRecordsRes.provisionalRecords[1].createdTimestamp
+						);
+		} else {
+			await Promise.reject('Tech record Search returned nothing.');
+		}
+	};
+
+	/**
+	 * helper function is used to process records and count provisional and current records
+	 * @param records
+	 */
+	private groupRecordsByStatusCode = (
```

Comment:
Might be one for the future, but there is now a `groupBy` method from ES2023 onwards (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy). 

We could have then had removed this method completely and utilised something like

`const { current, provisional } = Object.groupBy(records, (record) => record.techRecord_statusCode)`

Reply:
Absolutely agree 👍

Also agree that I think functional changes should be in follow up PRs.